code/modules/projectiles/projectile.dm

```diff
@@ -569,7 +569,8 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 
 /obj/projectile/proc/scan_a_turf(turf/turf_to_scan, cardinal_move)
 	if(turf_to_scan.density) //Handle wall hit.
-		ammo.on_hit_turf(turf_to_scan, src)
+		if((ammo.flags_ammo_behavior & (AMMO_EXPLOSIVE|AMMO_PASS_THROUGH_TURF)) != (AMMO_EXPLOSIVE|AMMO_PASS_THROUGH_TURF))
```

Comment:
Well, the & operator isn't exactly "has a flag". In case of checking for a singular flag, it works fine as such. However, lets say you check for a combined bitflag of (1<<0)|(1<<1) which translates to 3. If you check it for something like 0 it returns false and if you check 3 it returns true. Seems okay right? Well, lets check something that only has one of the two, e.g. (1<<1), or well, a 2. That also returns true. Why? Well, (1<<1) & (1<<0)|(1<<1) is 2, which by byond statement conditions is true. So we need to check for if both the bitfields are actually present, which in this case would be [input] & (1<<0)|(1<<1) == (1<<0)|(1<<1), aka, the returned value has to be 3, not 2 or 1, which it might return and let through otherwise.

In this case it's a != because anything except those with both flags are supposed to pass the check.

tl;dr checking for multiple bitflags in one check needs a == like this one to avoid false positives a simple & could otherwise cause.